module/Activity/src/Activity/Controller/AdminApprovalController.php

```diff
@@ -0,0 +1,227 @@
+        $disapprovedActivities = $queryService->getDisapprovedActivities();
+        $updatedActivities = [];
+        $updateProposals = [];
+        foreach ($queryService->getAllProposals() as $updateProposal) {
+            $updatedActivities[$updateProposal->getId()] = $updateProposal->getNew();
+            $updateProposals[$updateProposal->getNew()->getId()] = $updateProposal;
+        }
+        return [
+            'unapprovedActivities' => array_slice($unapprovedActivities, 0, $perPage),
+            'approvedActivities' => array_slice($approvedActivities, 0, $perPage),
+            'disapprovedActivities' => array_slice($disapprovedActivities, 0, $perPage),
+            'updatedActivities' => array_slice($updatedActivities, 0, $perPage),
+            'updateProposals' => $updateProposals,
+            'moreUnapprovedActivities' => count($unapprovedActivities) > $perPage,
+            'moreApprovedActivities' => count($approvedActivities) > $perPage,
+            'moreDisapprovedActivities' => count($disapprovedActivities) > $perPage
+        ];
+    }
+
+    /**
+     * View one activity.
+     */
+    public function viewAction()
+    {
+        $id = (int) $this->params('id');
+        $queryService = $this->getServiceLocator()->get('activity_service_activityQuery');
+
+        /** @var $activity Activity*/
+        $activity = $queryService->getActivity($id);
+
+        if (is_null($activity)) {
+            return $this->notFoundAction();
+        }
+
+        return [
+            'activity' => $activity,
+            'approvalForm' => new RequestForm('updateApprovalStatus', 'Approve'),
+            'disapprovalForm' => new RequestForm('updateApprovalStatus', 'Disapprove'),
+            'resetForm' => new RequestForm('updateApprovalStatus', 'Reset')
+        ];
+    }
+
+    /**
+     * Approve of an activity
+     */
+    public function approveAction()
+    {
+        return $this->setApprovalStatus('approve');
+    }
+
+    /**
+     * Disapprove an activity
+     */
+    public function disapproveAction()
+    {
+        return $this->setApprovalStatus('disapprove');
+    }
+
+    /**
+     * Reset the approval status of an activity
+     */
+    public function resetAction()
+    {
+        return $this->setApprovalStatus('reset');
+    }
+
+    /**
+     * Display the proposed update
+     */
+    public function viewProposalAction()
+    {
+        $id = (int) $this->params('id');
+        $queryService = $this->getServiceLocator()->get('activity_service_activityQuery');
+
+        $proposal = $queryService->getProposal($id);
+
+        if (is_null($proposal)) {
+            return $this->notFoundAction();
+        }
+
+        return [
+            'proposal' => $proposal,
+            'proposalApplyForm' => new RequestForm('proposalApply', 'Apply update'),
+            'proposalRevokeForm' => new RequestForm('proposalRevoke', 'Revoke update')
+            ];
```

Comment:
Curiously, codeclimate does not detect this one. Perhaps because this add is a result of a file rename.